aineko/extras/connectors/rest_poller.py

```diff
@@ -0,0 +1,260 @@
+    `{$SECRET_NAME}`. For example, if you have a secret named `SECRET_NAME`
+    that contains the value `SECRET_VALUE`, you can inject it into the url
+    field by passing `https://example.com?secret={$SECRET_NAME}` as the url.
+    The connector will then replace `{$SECRET_NAME}` with `SECRET_VALUE` before
+    connecting to the REST endpoint.
+
+    Example usage in pipeline.yml:
+    ```yaml title="pipeline.yml"
+    pipeline:
+      nodes:
+        RestPoller:
+          class: aineko.extras.RestPoller
+          outputs:
+            - test_rest
+          node_params:
+            url: "https://example.com"
+            headers:
+              auth: "Bearer {$SECRET_NAME}"
+            data: {"Greeting": "Hello, world!"}
+    ```
+
+    By default, this node will poll the REST endpoint every 5 seconds. This can
+    be changed by setting the `poll_interval` field in `node_params`. If the
+    REST endpoint is not expected to be available at all times, it is
+    recommended to increase `poll_interval` to reduce the number of requests
+    sent to the endpoint.
+
+    By default, this node will retry connecting to the REST endpoint forever
+    if the connection fails. This can be changed by setting the `max_retries`
+    field in `node_params`. If the REST endpoint is not expected to be
+    available at all times, it is recommended to set `max_retries` to a
+    finite number to prevent the node from retrying forever.
+
+    By default, this node will retry connecting to the REST endpoint every 5
+    seconds if the connection fails. This can be changed by setting the
+    `retry_sleep` field in `node_params`. If the REST endpoint is not expected
+    to be available at all times, it is recommended to increase `retry_sleep`
+    to reduce the number of requests sent to the endpoint.
+
+    By default, this node will consider any HTTP status code in the 200s to be
+    a success code. This can be changed by setting the `success_codes` field
+    in `node_params`. If the REST endpoint returns a non-200 status code on
+    success, it is recommended to add the status code to the `success_codes`
+    list.
+
+    By default, this node will not attach headers or data to the request. This
+    can be changed by setting the `headers` and `data` fields in `node_params`.
+    If the REST endpoint requires headers or data to be sent, it is
+    you should set the `headers` and `data` fields in `node_params`.
+
+    By default, this node will timeout after 10 seconds. This can be changed
+    by setting the `timeout` field in `node_params`. If the REST endpoint is
+    expected to take a long time to respond, it is recommended to increase
+    `timeout` to prevent the node from timing out.
+    """
+
+    # Poll settings
+    last_poll_time = time.time()
+    retry_count = 0
+
+    def _pre_loop_hook(self, params: Optional[Dict] = None) -> None:
+        """Initializes connection to API."""
+        # Cast params to ParamsREST type
```

Comment:
Personal opinion: comments like this don't add any value

Reply:
sure.  i tend to like to write comments such that 6 months from now i can scan comments and understand the functionality without fully understanding the code. since we're not really following this pattern everywhere, im fine removing this.